app/scripts/controllers/deploymentConfig.js

```diff
@@ -87,6 +95,33 @@ angular.module('openshiftConsole')
             $scope.deploymentConfig = deploymentConfig;
             updateHPAWarnings();
             ImageStreamResolver.fetchReferencedImageStreamImages([deploymentConfig.spec.template], $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
+            copyDeploymentConfigAndEnsureEnv(deploymentConfig);
+            $scope.saveEnvVars = function() {
+              // copy again to clean off any of angulars $$ vars from the template
```

Comment:
Is this needed? Angular toJSON strips these, which I'd guess is called under the hood when we make the HTTP request.

https://docs.angularjs.org/api/ng/function/angular.toJson